### PR TITLE
fix(cli): add support for components with non-default names (#10200)

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [ ] I have added a brief description of why this PR is necessary and/or what this PR solves.

Please see [Contribution FAQs](https://argo-cd.readthedocs.io/en/latest/developer-guide/faq/) if you have questions about your pull-request.

### CODE REVIEW

File: cmd/argocd/commands/admin/cluster.go [hunk 1]
```diff
@@ -68,7 +69,7 @@ type ClusterWithInfo struct {
 	Namespaces []string
 }
 
-func loadClusters(ctx context.Context, kubeClient *kubernetes.Clientset, appClient *versioned.Clientset, replicas int, namespace string, portForwardRedis bool, cacheSrc func() (*appstatecache.Cache, error), shard int) ([]ClusterWithInfo, error) {
+func loadClusters(ctx context.Context, kubeClient *kubernetes.Clientset, appClient *versioned.Clientset, replicas int, namespace string, portForwardRedis bool, cacheSrc func() (*appstatecache.Cache, error), shard int, redisName string, redisHaHaproxyName string) ([]ClusterWithInfo, error) {
```
Comment (ishitasequeira) [inline cmd/argocd/commands/admin/cluster.go:72]: rename the variable to `redisHaHaproxyName ` -> `redisHaproxyName`?

File: common/common.go [hunk 2]
```diff
@@ -138,6 +138,8 @@ const (
 	// LabelKeyAppInstance is the label key to use to uniquely identify the instance of an application
 	// The Argo CD application name is used as the instance name
 	LabelKeyAppInstance = "app.kubernetes.io/instance"
+	// LabelKeyAppName is the name of the Kubernetes application
+	LabelKeyAppName = "app.kubernetes.io/name"
```
Comment (ishitasequeira) [inline common/common.go:142]: We should change the comment here to  below or something similar?
```
LabelKeyAppName is the label key to use to uniquely identify name of the Kubernetes application
```

File: common/common.go [hunk 3]
```diff
@@ -233,6 +235,16 @@ const (
...
+	// EnvServerName is the pod selector labels of the Argo CD server component.
+	EnvServerName = "ARGOCD_SERVER_NAME"
+	// EnvRepoServerName is the pod selector labels of the Argo CD repo server component.
+	EnvRepoServerName = "ARGOCD_REPO_SERVER_NAME"
+	// EnvAppControllerName is the pod selector labels of the Argo CD application controller component.
+	EnvAppControllerName = "ARGOCD_APPLICATION_CONTROLLER_NAME"
+	// EnvRedisName is the pod selector labels of the Argo CD redis component.
+	EnvRedisName = "ARGOCD_REDIS_NAME"
+	// EnvRedisHaHaproxyName is the pod selector labels of the Argo CD Redis HA proxy component.
+	EnvRedisHaHaproxyName = "ARGOCD_REDIS_HAPROXY_NAME"
```
Comment (ishitasequeira) [inline common/common.go:247]: Should we represent in the comments that the env variable holds the custom name of the component and not the pod selector label?

**Review by ishitasequeira — CHANGES_REQUESTED** [review]

**Review by ishitasequeira — COMMENTED** [review]
Should we also add the same change for applicationset_controller as well?

**Comment by crenshaw-dev** [comment]
@ishitasequeira completed all the above! Can you detail more what changes you're thinking about for the applicationset controller?

**Comment by crenshaw-dev** [comment]
@ishitasequeira I don't think the Argo CD API client currently accepts an appset controller name, presumably because the client does not interact with that controller in any way yet.

**Comment by ishitasequeira** [comment]
> @ishitasequeira I don't think the Argo CD API client currently accepts an appset controller name, presumably because the client does not interact with that controller in any way yet.

That makes sense!! We can ignore the appset controller comment.

**Review by ishitasequeira — APPROVED** [review]
Thanks @crenshaw-dev, LGTM!!

**Comment by keskad** [comment]
Thanks!

**Pull request merged**